geomstats/_backend/pytorch/__init__.py

```diff
@@ -519,6 +519,10 @@ def equal(a, b, **kwargs):
     return torch.eq(a, b, **kwargs)
 
 
+def diag_indices(*args, **kwargs):
+    return tuple(map(torch.from_numpy, _np.diag_indices(*args, **kwargs)))
```

Comment:
We have been using the `torch.from_numpy` function a lot in this backend, ...but this is bad and was meant to be a hack: it might fail when it is used with pytorch tensors on GPUs (because numpy would not know how to compute on this).

To avoid this, could we get an implementation of diag_indices that is fully pytorch?

Many functions of the backends need a similar cleanup...

Reply:
So I working on other PR that updates pytorch version and clean the backend. They have added triu_indices (https://pytorch.org/docs/stable/generated/torch.triu_indices.html) and tril_indices (https://pytorch.org/docs/stable/generated/torch.tril_indices.html) in 1.8. Nothing has been added for diag_indices yet unfortunately. so I will remove diag_indices and find another way to get the result. And clean pytorch backend in the new PR.